webapp/channels/src/actions/views/root.ts

```diff
@@ -20,22 +34,56 @@ const pluginTranslationSources: Record<string, TranslationPluginFunction> = {};
 
 export type TranslationPluginFunction = (locale: string) => Translations
 
-export function loadConfigAndMe(): ThunkActionFunc<Promise<{config?: ClientConfig; isMeLoaded: boolean}>> {
-    return async (dispatch) => {
-        const results = await Promise.all([
-            dispatch(getClientConfig()),
-            dispatch(getLicenseConfig()),
-        ]);
-
-        let isMeLoaded = false;
-        if (document.cookie.includes('MMUSERID=')) {
-            const dataFromLoadMe = await dispatch(loadMe());
-            isMeLoaded = dataFromLoadMe?.data ?? false;
+/**
+ * This function meant to be used in root.tsx component loads config, license and if user is logged in, it loads user and its related data.
+ */
+export function loadConfigAndMe(): ThunkActionFunc<Promise<{isLoaded: boolean; isMeRequested?: boolean}>> {
+    return async (dispatch, getState) => {
+        // attempt to load config and license regardless if user is logged in or not
+        try {
+            await Promise.all([
+                dispatch(getClientConfig()),
+                dispatch(getLicenseConfig()),
+            ]);
+        } catch (error) {
+            dispatch(logError(error as ServerError));
+            return {
+                isLoaded: false,
+            };
+        }
+
+        // Return early if user is not logged in
+        if (!doesCookieContainsMMUserId()) {
+            return {
+                isLoaded: true,
+                isMeRequested: false,
+            };
+        }
+
+        // Load user and its related data now that we know that user is logged in
+        const serverVersion = getState().entities.general.serverVersion || Client4.getServerVersion();
+        dispatch({type: GeneralTypes.RECEIVED_SERVER_VERSION, data: serverVersion});
+
+        try {
+            await Promise.all([
+                dispatch(getMe()),
+                dispatch(getMyPreferences()),
+                dispatch(getMyTeams()),
+                dispatch(getMyTeamMembers()),
+            ]);
+
+            dispatch(getMyTeamUnreads(isCollapsedThreadsEnabled(getState())));
+            dispatch(getServerLimits());
+        } catch (error) {
+            dispatch(logError(error as ServerError));
+            return {
+                isLoaded: false,
+            };
         }
 
         return {
-            config: results[0].data,
-            isMeLoaded,
+            isLoaded: true,
+            isMeRequested: true,
```

Comment:
Did you still intend to rename `isMeLoaded`? Since we're awaiting on the actions above, `isMeRequested` still behaves the same as `isMeLoaded` did before

Reply:
Yes `isMeLoaded` is still need because if you see in the root.tsx code we are checking that to redirect to correct team etc. Open to other ideas as well